R/qenv-get_code.R

```diff
@@ -124,13 +122,9 @@ setMethod("get_code", signature = "qenv", function(object, deparse = TRUE, names
   }
 
   if (deparse) {
-    if (length(code) == 0) {
-      code
-    } else {
-      paste(code, collapse = "\n")
-    }
+    gsub(";\n", ";", paste(gsub("\n$", "", unlist(code)), collapse = "\n"))
```

Comment:
- remove "\n" to `paste(collapse = "\n")` is redundant or even wrong (it will remove "\n" from the last item). 
- also is `";\n"` a problem? It is a valid call


```suggestion
    paste(unlist(code), collapse = "")
```

Reply:
Ok, I'm wrong here. There is possibility that somebody provides a single call without terminator. The code will be preserved as is, for example `"a <- 1"`. Ignore my comment here ⬆️

Reply:
I experimented with this a bit before I pushed it

That's what we get with `paste(unlist(code), collapse = "")`

```r
q <- qenv()
q <- eval_code(q, quote(x <- 1))
q <- eval_code(q, quote(y <- x))
get_code(q)
[1] "x <- 1y <- x"
```

When we use `collapse = "\n"`, then this is fine

```
"x <- 1ny <- x"
```

but 

```r
q <- qenv()
q <- eval_code(q, quote(x <- 1))
q <- eval_code(
  q,
  quote({
    y <- x
    z <- 5
  })
)
```

gives 2 `\n`s after x

```r
> get_code(q)
[1] "x <- 1\ny <- x\n\nz <- 5"
```

```
> cat()
x <- 1
y <- x

z <- 5
```


Lastly, the `";\n", ";"` substitution is for cases like

```r
testthat::test_that("get_code returns code with comments and empty spaces", {
  code <- "
    # header comment after white space

    a <- 1L; b <- 2 #inline comment


    c <- 3
    # closing comment
    "
  q <- eval_code(qenv(), code)
  testthat::expect_equal(get_code(q), code)
})
```

<img width="389" alt="image" src="https://github.com/user-attachments/assets/8b258259-bd1c-4a15-8bb3-210cdf0e113f">


Those are first few tests in `test-qenv_get_code`.

We do actually a manual imputation for cases with `:` in the middle line. We add `\n`. that's why we need to deal with `;` and `\n` when extracting the code. Those are those lines https://github.com/insightsengineering/teal.code/pull/216/files#diff-66a61facc4f5ca86215d63af40e2645b364950439051784916316ef78d5073c8R476-R479

> it will remove "\n" from the last item

I think we can leave with a situation where the last one is used, but should not be. This is the case where it was the last one in the code. All other cases are tested